docs/contracts/functions.rst

```diff
@@ -139,12 +129,7 @@ If you use an early ``return`` to leave a function that has return variables,
 you must provide return values together with the return statement.
 
 .. note::
-    You cannot return some types from non-internal functions, notably
-    multi-dimensional dynamic arrays and structs. If you enable the
-    ABI coder v2 by adding ``pragma abicoder v2;``
-    to your source file then more types are available, but
-    ``mapping`` types are still limited to inside a single contract and you
-    cannot transfer them.
+    You cannot return some types from non-internal functions, notably ``mapping``.
```

Comment:
Stricly speaking, ``mapping``s themselves and any type that recursively contains some mapping (as struct member or array base type). Also strictly speaking, you only can't return them from non-internal functions except library functions (for external library functions it'll work at least theoretically).
But since this wasn't explained in full detail here before anyways, I'd also be fine with the suggested change.

Reply:
I'm glad to know the exact specifications. Thank you.